### DISCUSSION TITLE
`WikiOverlay`'s error page fails to redirect back to `Main_Page` occasionally

### DISCUSSION THREAD

**Discussion by Stedoss** [ppy/osu#21726] [Q&A]
Regressed in https://github.com/ppy/osu/pull/19246/commits/474c1a8a7a9f2a1d91d60182596604333890f202

Sorry for the nondescript title, hopefully I can explain it better. When a Wiki Page fails to load, it calls `WikiOverlay.onFail()` which does not set the `wikiData`'s path to a new one. This means that if a page linked by `Main_Page` doesn't resolve or errors, the resulting error page will not redirect when clicking the link because the new linked page is the same as the old one (I can't explain this very well can I...)

Reproduction steps:
1. Open `WikiOverlay` in game
2. Click on `Sitemap` in the `Wiki and meta pages` section
3. Error page will appear with a redirect link
4. Link will do nothing due to the early return in the linked commit

I tried to fix this myself but wasn't really sure if we should respect the failed request's path and set it as the `wikiData`'s current path, or have it set to an `Error` path, or something that I was just missing that would be easier.

**Comment by peppy** [comment]
If possible, could you please provide a video showing this happening?

**Reply by Stedoss** [reply]
Of course: 

https://user-images.githubusercontent.com/29103029/208630972-bdbd10ef-9ae3-4991-98f3-1d1478d50d02.mp4
